ui/imports/shared/popups/ImportCommunityPopup.qml

```diff
@@ -15,111 +15,103 @@ import StatusQ.Controls 0.1
 
 StatusDialog {
     id: root
+
     property var store
+    property alias text: keyInput.text
+
+    signal joinCommunityRequested(string communityId, var communityDetails)
+
     width: 640
     title: qsTr("Import Community")
 
-    signal joinCommunity(string communityId, var communityDetails)
-
     QtObject {
         id: d
         property string importErrorMessage
 
-        property bool loading
-        property bool communityFound: (d.communityDetails !== null && !!d.communityDetails.name)
-        property var communityDetails: {
-            if (!isInputValid) {
-                loading = false
-                return null
-            }
-            loading = true
-            const key = isPublicKey ? Utils.getCompressedPk(publicKey) :
-                                      root.store.getCommunityPublicKeyFromPrivateKey(inputKey, true /*importing*/);
-
-            const details = root.store.getCommunityDetails(key)
-            if (!!details) // the above can return `null` in which case we continue loading
-                loading = false
-            return details
-        }
+        property bool communityFound: !!d.communityDetails && !!d.communityDetails.name
+        property var communityDetails: null
+
+        property var requestedCommunityDetails: null
 
         readonly property string inputErrorMessage: isInputValid ? "" : qsTr("Invalid key")
         readonly property string errorMessage: importErrorMessage || inputErrorMessage
         readonly property string inputKey: keyInput.text.trim()
-        readonly property bool isPrivateKey: (Utils.isPrivateKey(inputKey))
-        readonly property bool isPublicKey: (publicKey !== "")
-        readonly property string privateKey: inputKey
         readonly property string publicKey: {
-            if (!Utils.isStatusDeepLink(inputKey)) {
-                const key = Utils.dropCommunityLinkPrefix(inputKey)
-                if (!Utils.isCommunityPublicKey(key))
-                    return ""
-                if (!Utils.isCompressedPubKey(key))
-                    return key
-                return Utils.changeCommunityKeyCompression(key)
-            } else {
-                return Utils.getCommunityDataFromSharedLink(inputKey).communityId;
+            if (Utils.isStatusDeepLink(inputKey)) {
+                const linkData = Utils.getCommunityDataFromSharedLink(inputKey)
+                return !linkData ? "" : linkData.communityId
```

Comment:
```suggestion
                return !!linkData ? linkData.communityId : ""
```

I always find this double negative idiom ugly :)

Reply:
I don't get it, you find it ugly, but still suggest? 😄